Formula/curl.rb

```diff
@@ -39,6 +42,13 @@ class Curl < Formula
   depends_on "c-ares" => :optional
   depends_on "libmetalink" => :optional
   depends_on "nghttp2" => :optional
+  depends_on "libunistring" if build.with? "libidn2"
```

Comment:
I thought about it. I just couldn't come to a decision I was that happy with either way 😅.

I loathe the `if build.with?` dependencies because Homebrew handles them sub-optimally, but at the same time `option "with-libunistring", "Build with IDN support"` or such feels like a stretch.